bundler/spec/bundler/fetcher_spec.rb

```diff
@@ -142,19 +142,20 @@
     end
 
     describe "include CI information" do
+      # We actually run these tests _on_ github actions, so these may be set during CI
+      let(:hide_vars) { { "CI" => nil, "GITHUB_ACTIONS" => nil, "CI_NAME" => nil } }
```

Comment:
This approach may be brittle to rubygems running on other CIs. What do you think of calculating the output before and after you add JENKINS_URL and checking if it includes the same set plus jenkins (possibly allowing it to already have jenkins)?

Or you could just revert to the match based test.

Reply:
Oh, it would be - I didn't realize that was a concern!

I could just replace all of ENV for those tests, instead of setting individual values with the helper.. but your tests do really on ci ENV quite a bit. I can't _think_ of any way these particular tests could be using one, but your suggestion will be safer.

I'll get that in shortly :-)

Reply:
Sorry, that rebase also pulled the master changes in, which makes the diff hard to see imo. Here's what I did:

```patch
diff --git a/bundler/spec/bundler/fetcher_spec.rb b/bundler/spec/bundler/fetcher_spec.rb
index 332ea81c7..f764c44fa 100644
--- a/bundler/spec/bundler/fetcher_spec.rb
+++ b/bundler/spec/bundler/fetcher_spec.rb
@@ -142,20 +142,21 @@
     end
 
     describe "include CI information" do
-      # We actually run these tests _on_ github actions, so these may be set during CI
-      let(:hide_vars) { { "CI" => nil, "GITHUB_ACTIONS" => nil, "CI_NAME" => nil } }
-
       it "from one CI" do
-        with_env_vars(hide_vars.merge("JENKINS_URL" => "foo")) do
+        with_env_vars("CI" => nil, "JENKINS_URL" => "foo") do
           ci_part = fetcher.user_agent.split(" ").find {|x| x.start_with?("ci/") }
-          expect(ci_part).to eq("ci/jenkins")
+          cis = ci_part.split("/").last.split(",")
+          expect(cis).to include("jenkins")
+          expect(cis).not_to include("ci")
         end
       end
 
       it "from many CI" do
-        with_env_vars(hide_vars.merge("TRAVIS" => "foo", "GITLAB_CI" => "gitlab", "CI_NAME" => "MY_ci")) do
+        with_env_vars("CI" => "true", "SEMAPHORE" => nil, "TRAVIS" => "foo", "GITLAB_CI" => "gitlab", "CI_NAME" => "MY_ci") do
           ci_part = fetcher.user_agent.split(" ").find {|x| x.start_with?("ci/") }
-          expect(ci_part).to eq("ci/gitlab,my_ci,travis")
+          cis = ci_part.split("/").last.split(",")
+          expect(cis).to include("ci", "gitlab", "my_ci", "travis")
+          expect(cis).not_to include("semaphore")
         end
       end
     end
```